app/helpers/mb_form_builder.rb

```diff
@@ -23,30 +23,51 @@ def mb_input_field(
       spellcheck: "false",
     }.merge(options)
 
-    if object.respond_to?(:hash_key_attribute?) &&
-        object.hash_key_attribute?(method)
+    field_values(text_field_options, method).map.with_index do |value, i|
+      if array_of_values?(method)
+        text_field_options[:name] = "#{object_name}[#{method}][]"
+        text_field_options[:id] = "#{object_name}_#{method}_#{i}"
+        options[:id] = "#{object_name}_#{method}_#{i}"
+      end
```

Comment:
if there is an array of values, the input name, id, and label id, should all be changed so that the html is correct.

eg:

```
name = "step[income][]"
id = step_income_1
label for = step_income_1
```

Reply:
oooh I think Christa and I dealt with a similar issue where we had to set name and id explicitly for a field and it was totally bananas.

see `app/views/medicaid/amounts_income/edit.html.erb`

Reply:
@jessieay I'm 99% sure that I would consider _*that*_ a bug. To work around the issue by setting attributes to get the expected behavior correct? That's definitely a bug.